### PR TITLE
Support in linux and add functional tests

### PR DESCRIPTION
**Pull request by sowu880**
1. Add functional tests for vscode client in OCIArtifactFunctionaltests.cs. Simulate different situations and listed in the inline comments at the top of the cs file.
2. Update functional pipeline, refine registry building stage in windows vm.
3. Support template management CLI in linux system and could be extend to other system in the future.

### CODE REVIEW

File: src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs [hunk 1]
```diff
@@ -93,5 +106,27 @@ private async Task OrasExecutionAsync(string command, string orasWorkingDirector
                 throw new OrasException(TemplateManagementErrorCode.OrasTimeOut, "Oras request timeout");
             }
         }
+
+        private void AddFilePermissionInLinuxSystem(string fileName)
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs:110]: Can we move this function to somewhere in common? It seems an useful function and used in unit tests as well.

File: src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs [hunk 2]
```diff
@@ -18,11 +16,25 @@ namespace Microsoft.Health.Fhir.TemplateManagement.Client
     public class OrasClient : IOrasClient
     {
         private readonly string _imageReference;
+        private readonly string orasFileName;
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs:19]: Same naming styles for private fields would be better.

File: src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/OCIFileManagerTests.cs [hunk 3]
```diff
@@ -18,28 +17,49 @@ public class OCIFileManagerTests
...
+            {
+                case PlatformID.Win32NT:
+                    _orasFileName = "oras-win.exe";
+                    break;
+                case PlatformID.Unix:
+                    _orasFileName = "oras-unix";
+                    OrasUtility.AddFilePermissionInLinuxSystem(_orasFileName);
+                    break;
+                default:
+                    throw new SystemException("System operation is not supported");
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/OCIFileManagerTests.cs:36]: Is this exception caught outside into TemplateMangementException?

File: release.yml [hunk 4]
```diff
@@ -127,7 +127,10 @@ stages:
         artifactName: FhirConverterBuild
         downloadPath: $(System.DefaultWorkingDirectory)
     - script: |
-        docker run --rm -d -p 5000:5000 --name registry stefanscherer/registry-windows:2.6.2
+        git clone -q https://github.com/sowu880/dockerfiles-windows.git
```
Comment (BoyaWu10) [inline release.yml:130]: Are we going to maintain our own version of windows image?

**Review by BoyaWu10 — COMMENTED** [review]
Let's update the title of the PR to a more formal one. The title will be included in release notes.

File: release.yml (hunk 4 above)
Comment (sowu880) [reply release.yml:130]: The public image can not be used for OCI artifacts.

**Comment by sowu880** [comment]
Separate it into two PR #156 and #157

**Pull request closed without merging**